### PR TITLE
Add module for SDC host

### PR DESCRIPTION
**Pull request by shekhar-j**
Add SDC deployment module for Linux. 

Tests:
Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Destroy complete! Resources: 4 destroyed.

### CODE REVIEW

File: examples/sdc_host_linux/main.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# /*
```
Comment (shenda1) [inline examples/sdc_host_linux/main.tf:1]: Can we include readme for examples as well?

File: examples/sdc_host_linux/variables.tf [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    url = string
+    pkg_name = string
+    local_dir = string
+    remote_dir = string
+    # use the SDC package on remote machine path (where SDC is deployed)
+    use_remote_path = bool
+  })
+  default = {
+    pkg_name = "emc-sdc-package.tar"
+    url = "http://example.com/EMC-ScaleIO-lia-4.0-0.687.Ubuntu.14.04.x86_64.tar" #"ftp://username:password@ftpserver/path/to/file"
```
Comment (doriac11) [inline examples/sdc_host_linux/variables.tf:80]: in the example should this be `sdc` instead of `lia`

**Review by nikitajoshi1 — APPROVED** [review]

**Review by doriac11 — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]

File: modules/sdc_host_linux/variables.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    password = string
+  })
+   default = {
+    user =  "root"
+    private_key = ""
+    certificate = ""
+    password = ""
+  }
+
+}
```
Comment (rounak-adhikary) [inline modules/sdc_host_linux/variables.tf:45]: Is `remote_host` entirely optional? If not, then:
1. It should not have a default.
2. Individual fields which are optional (eg. private_key) can be declared as optional. Ref: https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes.

Same question goes for `sdc_pkg`.

File: modules/sdc_host_linux/main.tf [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    type = "ssh"
+    user = self.output.user.name
+    private_key = self.output.user.private_key
+    certificate = self.output.user.certificate
+    host = self.output.ip
+    password = self.output.user.password
+  }
+  input = {
+      user = {
+          name = "root"
```
Comment (rounak-adhikary) [inline modules/sdc_host_linux/main.tf:78]: Can we use `var.remote_host.user` here? Applicable to other resources of this module.

File: modules/sdc_host_linux/variables.tf (hunk 3 above)
Comment (shekhar-j) [reply modules/sdc_host_linux/variables.tf:45]: thanks for that reference. I will update and verify.

**Review by rounak-adhikary — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**